src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -116,17 +112,14 @@ class RocksDbDataSource(
     this.handles = nameSpaces.zip(handles.toList).toMap
     this.dbOptions = dbOptions
     this.cfOptions = cfOptions
-    this.isClosed = false
   }
 
   /**
     * This function closes the DataSource, without deleting the files used by it.
     */
   override def close(): Unit = {
     logger.debug(s"About to close DataSource in path: ${rocksDbConfig.path}")
-    assureNotClosed()
-    isClosed = true
```

Comment:
I wonder if the solution to this problem should just be  calling `isClosed = true` inside the `try` block  after all close operations?